src/simix/smx_synchro.cpp

```diff
@@ -163,6 +163,8 @@ bool Mutex::try_lock(smx_actor_t issuer)
 void Mutex::unlock(smx_actor_t issuer)
 {
   XBT_IN("(%p, %p)", this, issuer);
+  if(!this->locked)
+    THROWF(mismatch_error, 0, "Cannot release that mutex: it was not locked.");
```

Comment:
I'd tend to think it'd be best to avoid using "XBT C-style exceptions" in new code.